src/nm-checkpoint.c

```diff
@@ -260,6 +260,12 @@ restore_and_activate_connection (NMCheckpoint *self,
 		need_activation = TRUE;
 	}
 
+	if (!dev_checkpoint->device) {
+		nm_assert (dev_checkpoint->is_software);
```

Comment:
where does this assert come from? I don't see why this is correct. Could you add a comment explaining why it's here?

Reply:
OK, I can add a comment there. Does the rest looks good?